### PR TITLE
Adds additional go IT and utils

### PR DESCRIPTION
**Pull request by nerdalert**
Signed-off-by: Brent Salisbury <bsalisbu@redhat.com>

### CODE REVIEW

File: internal/client/user.go [hunk 1]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
 
 	return u, nil
 }
+
+// PatchAddUserToZone patches a user into a zone
```
Comment (dave-tucker) [inline internal/client/user.go:50]:
```go
// AddUserToZone adds the current user to the given zone
```

File: internal/client/user.go [hunk 2]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+func (c *Client) PatchAddUserToZone(zoneID string) (models.PatchUser, error) {
```
Comment (dave-tucker) [inline internal/client/user.go:51]:
```suggestion
// MoveCurrentUserToZone moves the current user into a given zone
func (c *Client) MoveCurrentUserToZone(zoneID uuid.UUID) (models.User, error) {
```

Returns a `User`. https://github.com/redhat-et/apex/blob/main/internal/handlers/user.go#L47

File: internal/client/user.go [hunk 3]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+	dest := c.baseURL.JoinPath(CURRENT_USER).String()
+	zoneUUID, _ := uuid.Parse(zoneID)
+	user := models.PatchUser{
+		ZoneID: zoneUUID,
+	}
+	userJSON, _ := json.Marshal(user)
+
+	req, err := http.NewRequest(http.MethodPatch, dest, bytes.NewBuffer(userJSON))
+	if err != nil {
+		return models.PatchUser{}, err
```
Comment (dave-tucker) [inline internal/client/user.go:61]:
```suggestion
		return models.User{}, err
```

File: internal/client/user.go [hunk 4]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+	}
+
+	res, err := c.do(req)
+	if err != nil {
+		return models.PatchUser{}, err
```
Comment (dave-tucker) [inline internal/client/user.go:66]:
```suggestion
		return models.User{}, err
```

File: internal/client/user.go [hunk 5]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return models.PatchUser{}, fmt.Errorf("failed to patch the user into the zone: %s", zoneID)
```
Comment (dave-tucker) [inline internal/client/user.go:72]:
```suggestion
		return models.User{}, fmt.Errorf("failed to patch the user into the zone: %s", zoneID)
```

File: internal/client/user.go [hunk 6]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+		return models.PatchUser{}, err
+	}
+
+	res, err := c.do(req)
+	if err != nil {
+		return models.PatchUser{}, err
+	}
+
+	defer res.Body.Close()
+
```
Comment (dave-tucker) [inline internal/client/user.go:70]:
```suggestion

        body, err := io.ReadAll(res.Body)
	if err != nil {
		return models.User{}, err
	}
```

File: internal/client/user.go [hunk 7]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+	if res.StatusCode != http.StatusOK {
+		return models.PatchUser{}, fmt.Errorf("failed to patch the user into the zone: %s", zoneID)
+	}
+
+	return models.PatchUser{}, nil
```
Comment (dave-tucker) [inline internal/client/user.go:75]:
```suggestion
        var u models.User
	if err := json.Unmarshal(body, &u); err != nil {
		return models.User{}, err
	}
	
	return u, nil
```

File: internal/client/user.go [hunk 8]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
 
 	return u, nil
 }
+
+// PatchAddUserToZone patches a user into a zone
+func (c *Client) PatchAddUserToZone(zoneID string) (models.PatchUser, error) {
+	dest := c.baseURL.JoinPath(CURRENT_USER).String()
+	zoneUUID, _ := uuid.Parse(zoneID)
```
Comment (dave-tucker) [inline internal/client/user.go:53]:
```suggestion
```

File: internal/client/user.go [hunk 9]
```diff
@@ -44,3 +46,31 @@ func (c *Client) GetCurrentUser() (models.User, error) {
...
+	user := models.PatchUser{
+		ZoneID: zoneUUID,
```
Comment (dave-tucker) [inline internal/client/user.go:55]:
```suggestion
		ZoneID: zoneID,
```

**Review by dave-tucker — CHANGES_REQUESTED** [review]
Couple of changes - should be able to accept/commit these in the UI if gofmt approves.

**Comment by nerdalert** [comment]
> Couple of changes - should be able to accept/commit these in the UI if gofmt approves.

@dave-tucker Thanks! Looks green.

**Review by dave-tucker — APPROVED** [review]

**Pull request merged**